src/dbus.c

```diff
@@ -59,7 +65,17 @@ register_dbus (GDBusConnection  *connection,
     "       <arg direction=\"in\" type=\"s\" name=\"application_name\"/>"
     "       <arg direction=\"in\" type=\"s\" name=\"lock_file\"/>"
     "       <arg direction=\"in\" type=\"a{sv}\" name=\"extra_parameters\"/>"
-
+    "        </method>"
+    "       <!--"
+    "            ApplicationRefreshCompleted:"
+    "            @application_name: the name of the application name"
+    "           @extra_parameters: a dictionary with extra optional parameters."
+    "           Method used to notify the daemon that the refresh is completed"
+    "           and the dialog can be closed."
+    "       -->"
+    "       <method name=\"ApplicationRefreshCompleted\">"
+    "       <arg direction=\"in\" type=\"s\" name=\"application_name\"/>"
+    "       <arg direction=\"in\" type=\"a{sv}\" name=\"extra_parameters\"/>"
     "        </method>"
```

Comment:
Rather than embedding the D-Bus introspection data like this, it'd be worth moving it to its own file and generating skeletons to plug the implementation into. This can be done using meson's gnome.gdbus_codegen() helper:

https://mesonbuild.com/Gnome-module.html#gnomegdbus_codegen

The skeleton code will then handle method dispatch, and we'd have a little type safety ensuring the method implementations match the signatures.

That's probably worth doing as a separate PR though, given that it is a cleanup for what's already in place.

Reply:
Good idea!